src/middleware/dynamic_prompt.py

```
@@ -0,0 +1,24 @@
+
+if TYPE_CHECKING:
+    from langchain.agents.middleware import ModelRequest
+
+
+def create_dynamic_prompt_middleware(template: str):
+    @dynamic_prompt
+    def render_prompt(request: ModelRequest) -> str:
+        if not isinstance(request.runtime.context, AgentContext):
+            raise TypeError(
+                f"Expected AgentContext, got {type(request.runtime.context)}"
+            )
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)